Add .dds DX10 header support

stijnherfst opened the pull request:
When the FourCC of a .dds is DX10, there is an extended header specifying what underlying representation is used. This PR adds support for BC5 unorm when it is encapsulated in a DX10 header. Furthermore, the CMakeLists is simplified and updated so that it can now be easily used with the [vcpkg](https://github.com/microsoft/vcpkg/) package manager.

SpartanJ commented:
Excellent contribution! I'm just going to ask you to please also upload a test DX10 DDS image into the `bin` folder so it can be tested easily (may be just export `img_test.png` to DX10 DDS with the name `img_test_dx10.dds`).

Thanks!

stijnherfst commented:
I have added a DX10_BC5 test image and renamed the images in `/bin` to match their formats. I have also updated the README to clarify the capabilities for each of the BCx formats.

SpartanJ commented:
Awesome! Thanks!
